lib/Sema/ConstraintSystem.h

```diff
@@ -3071,6 +3071,35 @@ class ConstraintSystem {
     return typeVar->getImpl().getRepresentative(getSavedBindings());
   }
 
+  /// Find if the given type variable is representative for a type
+  /// variable which last locator path element is of the specified kind.
+  /// If true returns the type variable which it is the representative for.
+  TypeVariableType *
+  isRepresentativeFor(TypeVariableType *typeVar,
+                      ConstraintLocator::PathElementKind kind) const {
```

Comment:
One last thing (I promise) - could you please move implementation into `ConstraintSystem.cpp`?

Reply:
Sure! No problem :) 
I actually thought of that, but I've seen some other methods inline on the headers so I'd go for that. What are the criteria/rules of thumb we use normally for those cases?

Reply:
I think in general we'd like to inline only small methods but we have been really inconsistent about this...